core/templates/dev/head/domain/collection/CollectionObjectFactorySpec.js

```diff
@@ -324,6 +337,117 @@ describe('Collection object factory', function() {
+    expect(_addCollectionSkill('new skill')).toBe(true);
+    var newSkillId = _sampleCollection.getSkillIdFromName('new skill');
+    expect(newSkillId).toEqual('skill0');
+    expect(_sampleCollection.containsCollectionSkill(newSkillId)).toBe(true);
+
+    expect(_addCollectionSkill('new skill 2')).toBe(true);
+    var newSkillId2 = _sampleCollection.getSkillIdFromName('new skill 2');
+    expect(newSkillId2).toEqual('skill1');
+    expect(_sampleCollection.containsCollectionSkill(newSkillId2)).toBe(true);
+  });
+
+  it('should be able to delete skills', function() {
+    // Add collection nodes
+    _addCollectionNode('exp_id0');
+    _addCollectionNode('exp_id1');
+
+    var collectionNode0 = _getCollectionNode('exp_id0');
+    var collectionNode1 = _getCollectionNode('exp_id1');
+
+    // Add skills
+    expect(_addCollectionSkill('skill01')).toBe(true);
+    var skillId1 = _sampleCollection.getSkillIdFromName('skill01');
+    expect(_addCollectionSkill('skill02')).toBe(true);
+    var skillId2 = _sampleCollection.getSkillIdFromName('skill02');
+    expect(_sampleCollection.containsCollectionSkill(skillId2)).toBe(true);
+
+    collectionNode0.getAcquiredSkillList().addSkill(skillId2);
+    collectionNode1.getPrerequisiteSkillList().addSkill(skillId2);
+
+    // Delete a skill
+    expect(_sampleCollection.deleteCollectionSkill(skillId2)).toBe(true);
+    expect(_sampleCollection.containsCollectionSkill(skillId2)).toBe(false);
+
+    // Other skill still exists
+    expect(_sampleCollection.containsCollectionSkill(skillId2)).toBe(false);
+
+    // Skill ID is deleted from prerequisite and acquired skills
+    expect(collectionNode0.getAcquiredSkillList().getSkills()).toEqual([]);
+    expect(collectionNode1.getPrerequisiteSkillList().getSkills()).toEqual([]);
+
+    // Skill ID is not reused
+    expect(_addCollectionSkill('skill03')).toBe(true);
+    var skillId3 = _sampleCollection.getSkillIdFromName('skill03');
+    expect(skillId3).toEqual('skill2');
+  });
+
+  it('should be able to get all skills and clear skills', function() {
+    // Add skills.
+    expect(_addCollectionSkill('skill01')).toBe(true);
+    var skillId1 = _sampleCollection.getSkillIdFromName('skill01');
+    expect(_addCollectionSkill('skill02')).toBe(true);
+    var skillId2 = _sampleCollection.getSkillIdFromName('skill02');
+    expect(_sampleCollection.containsCollectionSkill(skillId2)).toBe(true);
+
+    // Check get all skills.
+    var expectedSkills = {};
+    expectedSkills[skillId1] = CollectionSkillObjectFactory.createFromIdAndName(
+        skillId1, 'skill01');
+    expectedSkills[skillId2] = CollectionSkillObjectFactory.createFromIdAndName(
+        skillId2, 'skill02');
+    expect(_sampleCollection.getCollectionSkills()).toEqual(expectedSkills);
+
+    // Clear skills.
+    _sampleCollection.clearCollectionSkills();
+    expect(_sampleCollection.getCollectionSkills()).toEqual({});
+
+    // Check that IDs reset.
```

Comment:
I don't think so, this should be part of checking that clearing worked correctly.